### ISSUE TITLE
Cannot train after setting up RASA container

### ISSUE THREAD

**Issue by DeadEnded** [synesthesiam/rhasspy-hassio-addon#46]
Trying to test out the Rasa NLU intent recognition.
I followed the instructions on Rasa sight and got the container up and going.
If I navigate to port 5005 on the host, I get:
```Hello from Rasa: 1.6.0a1```
So I think I got it working correctly.

But when I try to train in Rhasspy after changing the intent recognizer to Rasa (and setting the IP/port) I get this:

```
[ERROR:65466427] __main__: Training failed: JSONDecodeError('Expecting value: line 1 column 1 (char 0)',)
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/quart/app.py", line 1471, in full_dispatch_request
    result = await self.dispatch_request(request_context)
  File "/usr/local/lib/python3.6/dist-packages/quart/app.py", line 1519, in dispatch_request
    return await handler(**request_.view_args)
  File "app.py", line 514, in api_train
    raise Exception(f"Training failed: {result.reason}")
Exception: Training failed: JSONDecodeError('Expecting value: line 1 column 1 (char 0)',)
```

I haven't looking into the code yet to see if I understand what the issue is, but hope to later this week (I'm a novice though).  Creating the issue here so if someone more capable has time to look, they can fix it or tell me what I am doing wrong :-).

If there is more information I can supply let me know.

Cheers!
DeadEnd

**Comment by koenvervloesem** [comment]
Can you take a look at the logs of the Rasa container? It should show some decent log messages about what is going on.

**Comment by DeadEnded** [comment]
Thanks... apologies for not checking that beforehand... oops.

Here is what I see in the Rasa container log (I haven't adjusted logging yet):

```
2019-12-03 22:37:40 INFO     root  - Starting Rasa server on http://localhost:5005,
2019-12-03 22:37:44.576447: E tensorflow/stream_executor/cuda/cuda_driver.cc:318] failed call to cuInit: UNKNOWN ERROR (303),
Exception occurred while handling uri: 'http://192.168.0.10:5005/model/train?%7B%22project%22:%20%22rhasspy%22%7D',
Traceback (most recent call last):,
  File "/build/lib/python3.6/site-packages/sanic/app.py", line 918, in handle_request,
    handler, args, kwargs, uri = self.router.get(request),
  File "/build/lib/python3.6/site-packages/sanic/router.py", line 407, in get,
    return self._get(request.path, request.method, ""),
  File "/build/lib/python3.6/site-packages/sanic/router.py", line 470, in _get,
    raise NotFound("Requested URL {} not found".format(url)),
sanic.exceptions.NotFound: Requested URL /model/train not found,
```
Being in a container - I do not believe the graphics driver (if I recognize cuda correctly) is usable... not sure about it though.  The unknown exception happens when I tried to train.

I also navigated to `192.168.0.10:5005` and get a similar error about a missing icon.  So I think maybe there is part of the installation that was not completed?  I'm going to go back through the installation instructions to see if I missed something...

**Comment by koenvervloesem** [comment]
@DeadEnded Did you find the cause of this problem?

**Comment by DeadEnded** [comment]
Unfortunately I have been to busy to dig into it at all.
Life has been busy :-)

**Comment by xcmb-de** [comment]
Same Problem 

docker container: 
- synesthesiam/rhasspy-server:latest (2.4.15)
- rasa/rasa_nlu:0.14.4-full

rasa-nlu Log:
POST /model/train?%7B%22project%22:%20%22rhasspy%22%7D HTTP/1.1" 404 233 "-" "python-requests/2.22.0"

Found [RASA-NLU HTTP API ENDPOINTS](https://rasa.com/docs/rasa/nlu/using-nlu-only/)

training_url "train" instead "model/train"

"POST /train?%7B%22project%22:%20%22rhasspy%22%7D HTTP/1.1" 404 46 "-" "python-requests/2.22.0"

next Problem: wrong request parameter 
"project=rhasspy" without curved brackets, quotes and colon

after that hard test fixes:

"POST /train?project=rhasspy HTTP/1.1" 500 76 "-" "python-requests/2.22.0"
WARNING  rasa_nlu.data_router  - [Failure instance: Traceback (failure with no frames): <class 'rasa_nlu.train.TrainingException'>: Unknown data format for file /tmp/tmpwproutas_training_data

tmpwproutas_training_data content looks like a mix between md and json format in:
[Training Data Format](https://rasa.com/docs/rasa/nlu/training-data-format/)

and the sample files from rasa-nlu:
 /app/sample_configs/config_train_server_json.yml
 /app/sample_configs/config_train_server_md.yml

**Comment by synesthesiam** [comment]
Fixed in 2.4.19
